src/chrony.py

```diff
@@ -232,6 +256,98 @@ def write_config(self, config: str) -> None:
         """
         self.CONFIG_FILE.write_text(config, encoding="utf-8")  # pragma: nocover
 
+    def _make_certs_dir(self) -> None:  # pragma: nocover
+        """Create the chrony TLS certificates directory."""
+        self.CERTS_DIR.mkdir(exist_ok=True, mode=0o700)
+        shutil.chown(self.CERTS_DIR, "_chrony", "_chrony")
+
+    def _iter_certs_dir(self) -> list[pathlib.Path]:  # pragma: nocover
+        """Iterate over all certificate files in the certificate directory.
+
+        Returns:
+            An iterator over the paths of the certificate files.
+        """
+        return [f for f in self.CERTS_DIR.iterdir() if f.suffix in {".crt", ".key"}]
+
+    @staticmethod
+    def _write_certs_file(path: pathlib.Path, content: str) -> None:  # pragma: nocover
+        """Write content of a certificate file and set appropriate permissions and ownership.
+
+        Args:
+            path: The path to the certificate file.
+            content: The content to write to the file.
+        """
+        path.touch(mode=0o600, exist_ok=True)
+        path.write_text(content, encoding="utf-8")
+        shutil.chown(path, "_chrony", "_chrony")
+
+    @staticmethod
+    def _read_certs_file(path: pathlib.Path) -> str:
+        """Read and return the content of a certificate file.
+
+        Args:
+            path: The path to the certificate file.
+
+        Returns:
+            The content of the certificate file as a string.
+        """
+        return path.read_text(encoding="utf-8")  # pragma: nocover
+
+    @staticmethod
+    def _unlink_certs_file(path: pathlib.Path) -> None:
+        """Unlink (delete) a certificate file.
+
+        Args:
+            path: The path to the certificate file to delete.
+        """
+        path.unlink()  # pragma: nocover
```

Comment:
Just out of curiosity, should this have `path.unlink(missing_ok=True)` for idempotency?

Reply:
Yes, indeed, updated, thanks!